### PR TITLE
MutationPoint: dump mutation's source file context

### PR DESCRIPTION
**Pull request by stanislaw**

### CODE REVIEW

File: include/mull/Reporters/SourceManager.h [hunk 1]
```diff
@@ -18,6 +18,7 @@ struct LineOffset {
 class SourceManager {
 public:
   std::string getLine(const SourceLocation &location);
+  size_t getLinesNumber(const SourceLocation &location);
```
Comment (AlexDenisov) [inline include/mull/Reporters/SourceManager.h:21]: Why plural?

File: lib/Reporters/SourceManager.cpp [hunk 2]
```diff
@@ -12,7 +12,7 @@ SourceManager::~SourceManager() {
 
 std::string SourceManager::getLine(const SourceLocation &location) {
   LineOffset &lineOffset = getLineOffset(location);
-  assert(location.line < lineOffset.offsets.size());
+  assert(location.line <= lineOffset.offsets.size());
```
Comment (AlexDenisov) [inline lib/Reporters/SourceManager.cpp:15]: Could you elaborate on this? Accessing `size()` element would be out of bounds, right?

File: include/mull/Reporters/SourceManager.h (hunk 1 above)
Comment (stanislaw) [reply include/mull/Reporters/SourceManager.h]: Changed to `getNumberOfLines`.

File: lib/Reporters/SourceManager.cpp (hunk 2 above)
Comment (stanislaw) [reply lib/Reporters/SourceManager.cpp]: As discussed in person, the issue was related to the fact that the code didn't support getting the information for last lines in files.

**Pull request merged**